src/util/WalletBackendAdapter.ts

```diff
@@ -210,13 +291,29 @@ export const makeWalletBridgeFromFollowers = (
     }
     const currentEl: ValueFollowerElement<CurrentWalletRecord> = latest.value;
     const wallet = currentEl.value;
-    console.log('wallet current', wallet);
+    console.debug('wallet current', wallet);
+
+    const agoricBrands = await fetchAgoricBrands();
+    assert(agoricBrands, 'Failed to fetch agoric brands');
+
     for (const purse of wallet.purses) {
-      console.debug('registering purse', purse);
-      const brandDescriptor = wallet.brands.find(
-        bd => purse.brand === bd.brand,
-      );
-      assert(brandDescriptor, `missing descriptor for brand ${purse.brand}`);
+      // We only care about the zoe invite purse, which has asset kind 'set'.
+      // Non 'set' amounts need to be fetched from vbank to know their
+      // decimalPlaces.
+      //
+      // If we have add non 'set' amount purses that aren't in the vbank, it's
+      // not currently possible to read their decimalPlaces, so this code
+      // will need updating.
```

Comment:
This documentation is good.

Ideally we'd import some `invitationIssuerAssetKind` constant from `@agoric/zoe/something`.

I leave it to @turadg to judge the cost-effectiveness of that.

Reply:
I don't know of any runtime features of an invitation issuer that distinguishes it from any other Set kind issuer.

You could make this "set"-ness check more apparent using `@agoric/patterns` (soon to be in Endo): `matches(purse.balances.value, SetValueShape)`.

Reply:
Looks like `@agoric/patterns` doesn't exist yet either, but that would be cool.

Reply:
Oops I meant '@agoric/store'. But yeah let's wait for it to be in Endo